### PR TITLE
v7: Disable GraphQL field suggestions

### PR DESCRIPTION
**Pull request by jomunker**
Updates `app.module.ts` to disable GraphQL field suggestions in production. The added property looks like this:

```ts
// ...
formatError: (error) => {
    // Disable GraphQL field suggestions in production
    if (process.env.NODE_ENV !== "development") {
        if (error instanceof ValidationError) {
            return new ValidationError("Invalid request.");
        }
    }
    return error;
},
// ...
```

It also adds the type `ApolloDriverConfig` as generic to the `GraphQLModule.forRootAsync` to improve typing.

---

COM-493

### CODE REVIEW

**Comment by johnnyomair** [comment]
@nsams @jomunker created a more sophisticated upgrade script using ts-morph to perform the AST modifications. What do you think about it?

**Comment by nsams** [comment]
looking good, ts-morph is really relatively easy to use and a perfect fit for this upgrade

File: src/v7/hide-graphql-field-suggestions.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { Project, SyntaxKind } from "ts-morph";
+
+/**
+ * Adds the `formatError` property to `GraphQLModule` options to hide `GraphQL` field suggestions for non dev environments.
+ */
+export default async function hideGraphqlFieldSuggestions() {
+    const project = new Project({ tsConfigFilePath: "./api/tsconfig.json" });
```
[the remark this answers is not in the document]
Comment (jomunker) [reply src/v7/hide-graphql-field-suggestions.ts:7]: How do you mean that?

File: src/v7/hide-graphql-field-suggestions.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    });
+
+    // Get the AppModule class
+    const appModuleClass = sourceFile.getClassOrThrow("AppModule");
+
+    // Get the forRoot method within AppModule
+    const forRootMethod = appModuleClass.getMethodOrThrow("forRoot");
+
+    // Get the GraphQLModule.forRootAsync call within the forRoot method
+    const graphqlForRootCall = forRootMethod
```
[the remark this answers is not in the document]
Comment (jomunker) [reply src/v7/hide-graphql-field-suggestions.ts:30]: ✅

File: src/v7/hide-graphql-field-suggestions.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            }
+            return error;
+        }`;
+
+    // return if property formatError already exists
+    if (graphqlForRootCall.getArguments().find((arg) => arg.getText().includes("formatError"))) {
+        console.warn(
+            `WARNING: formatError property already exists in GraphQLModule.forRootAsync options. To be sure GraphQl field suggestions are disabled for non dev environments, please check if the implementation already contains: ${formatErrorImplText}`,
+        );
+        return;
```
[the remark this answers is not in the document]
Comment (jomunker) [reply src/v7/hide-graphql-field-suggestions.ts:59]: ✅

File: src/v7/hide-graphql-field-suggestions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnnyomair) [reply src/v7/hide-graphql-field-suggestions.ts:7]: I was thinking about initializing ts-morph with something like `new Project({ root: "api/" })`, but that doesn't seem possible.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**